routes/users_id.js

```diff
@@ -0,0 +1,61 @@
+const datastore = require('../modules/datastore.js');
+
+const USER_KIND = 'User';
+const SOLUTION_KIND = 'Solution';
+const PROBLEM_KIND = 'Problem';
+
+module.exports = function(app) {
+  async function getUser(username) {
+    const query = datastore
+        .createQuery(USER_KIND)
+        .filter('username', '=', username)
+        .limit(1);
+    const [users] = await datastore.runQuery(query);
+    return users[0];
+  }
+
+  /* Helper function to get solutions submitted by a user.
+   * TODO(b/161439920) filter by user ID rather than email. 
+   */
+  async function listSolutionsByUser(user) {
+    const query = datastore
+        .createQuery(SOLUTION_KIND)
+        .filter('email', '=', user.email);
+    const [userSolutions] = await datastore.runQuery(query);
+    return userSolutions;
+  }
+
+  async function getProblems(problemIds) {
```

Comment:
Maybe we should start considering some refactors with our Datastore queries.
Keeping them as a separate module or part of Datastore module might work to avoid code re-use.

Reply:
yeah, definitely agree